packages/amplication-server/src/core/project/project.service.ts

```diff
@@ -114,13 +120,64 @@ export class ProjectService {
     return [...changedEntities, ...changedBlocks];
   }
 
+  async validateSubscriptionPlanLimitationsForProject(
+    projectId: string
+  ): Promise<void> {
+    const project = await this.findUnique({
+      where: { id: projectId },
+    });
+    const workspace = await this.workspaceService.getWorkspace({
+      where: { id: project.workspaceId },
+    });
+    const subscription = await this.subscriptionService.getCurrentSubscription(
+      workspace.id
+    );
+
+    if (!subscription) {
+      const services = await this.resourceService.getWorkspaceServices(
+        workspace.id
+      );
+
+      if (services.length > 3) {
+        throw new Error(
+          "You have reached the maximum number of services. (Upgrade your workspace plan)"
+        );
+      }
+
+      const projectServices = await this.resourceService.resources({
+        where: { projectId: project.id },
+      });
+      const promises = projectServices.map((project) =>
+        this.validateSubscriptionPlanLimitationsForService(project)
+      );
```

Comment:
use a query directly on Prisma to find if there are any services in the project in the workspace with more then X entities